espflash/src/connection/mod.rs

```diff
@@ -277,6 +281,39 @@ impl Connection {
                 info!("Staying in flasher stub");
                 Ok(())
             }
+            ResetAfterOperation::WatchdogReset => {
+                info!("Resetting device with watchdog");
+
+                match chip {
+                    Chip::Esp32c3 => {
+                        wdt_reset(chip, self)?;
+                    }
+                    Chip::Esp32s2 => {
+                        let esp32s2 = esp32s2::Esp32s2;
```

Comment:
You can use [match binding](https://doc.rust-lang.org/rust-by-example/flow_control/match/binding.html)'s to avoid this temporary variable

Reply:
Ah sorry I missed that `esp32s2::Esp32s2` is a struct, not the enum we're matching on. Ignore me.